usecase/cooking/test/test_usecase.py

```diff
@@ -25,15 +30,27 @@ def setUpClass(self):
             self.calendar_service.set_remote(self.calendar_remote)
         else:
             print("Mocking remotes...")
-            self.use_case = Cook(True)
-            self.todoist_service = TodoistService.instance()
-            self.todoist_service.set_remote(TodoistMockRemote())
+            self.use_case = Cook()
+            
             self.calendar_remote = CaldavMockRemote()
             self.calendar_service = CalService.instance()
             self.calendar_service.set_remote(self.calendar_remote)
+            self.use_case.cal_service = self.calendar_service
+
+            self.yelp_service = YelpService.instance()
+            self.yelp_service.set_remote(YelpMock())
+            self.use_case.yelp_service = self.yelp_service
+
+            self.todoist_service = TodoistService.instance()
+            self.todoist_service.set_remote(TodoistMockRemote())
+            self.use_case.todoist_service = self.todoist_service
 
-    def test_trigger_usecase(self):
-        self.use_case.trigger_use_case('pork')
+            self.spoonacle_service = SpoonacularService.instance()
+            self.spoonacle_service.set_remote(SpoonacularMOCKRemote())
+            self.use_case.spoonacle_service = self.spoonacle_service
+            
+    def test_usecase(self):
+        self.use_case.advance({'ingredient': 'pork'})
```

Comment:
why don't you take the return of `advance` as a reply?

you could insert an assertIsInstance(reply, Reply)

are there any other asserts that could validate the correct behavior of the test? E.g. a certain substring that is contained in the reply?

Reply:
done